packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/service/SharedPreferencesNotificationCategoriesStore.java

```diff
@@ -0,0 +1,181 @@
+    return editor.remove(preferencesNotificationCategoryKey(identifier));
+  }
+
+  /**
+   * Removes all categories, returning removed IDs.
+   */
+  public Collection<String> removeAllNotificationCategories() {
+    Collection<String> deletedCategories = new ArrayList<>();
+    SharedPreferences.Editor editor = mSharedPreferences.edit();
+    for (NotificationCategory category : getAllNotificationCategories()) {
+      String identifier = category.getIdentifier();
+      removeNotificationCategory(editor, identifier);
+      deletedCategories.add(identifier);
+    }
+    boolean successful = editor.commit();
+    return successful ? deletedCategories : null;
+  }
+
+  /**
+   * Serializes the category to a Base64-encoded string.
+   *
+   * @param notificationCategory Notification category to serialize
+   * @return Base64-encoded, serialized category
+   * @throws IOException Thrown if there is an error while writing category to string.
+   */
+  private String serializeNotificationCategory(NotificationCategory notificationCategory) throws IOException {
+    try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+         ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream)) {
+      objectOutputStream.writeObject(notificationCategory);
+      return Base64.encodeToString(byteArrayOutputStream.toByteArray(), Base64.NO_WRAP);
+    }
+  }
+
+  /**
+   * Deserializes the category from the string representation.
+   *
+   * @param category Base64-encoded, serialized category representation
+   * @return Deserialized category or null
+   * @throws IOException            Thrown if there is an error while reading category from String
+   * @throws ClassNotFoundException Thrown if the deserialization failes due to class not being found.
+   * @throws InvalidClassException  Thrown if the category is of invalid class.
+   */
+  private NotificationCategory deserializeNotificationCategory(String category) throws IOException, ClassNotFoundException, InvalidClassException {
+    if (category == null) {
+      return null;
+    }
+    byte[] data = Base64.decode(category, Base64.NO_WRAP);
+    try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(data);
+         ObjectInputStream ois = new ObjectInputStream(byteArrayInputStream)) {
+      Object o = ois.readObject();
+      if (o instanceof NotificationCategory) {
+        return (NotificationCategory) o;
+      }
+      throw new InvalidClassException("Expected serialized notification category to be an instance of NotificationCategory. Found: " + o.toString());
+    }
+  }
+
+  /**
+   * @param identifier Category identifier
+   * @return Key under which the notification category will be persisted in storage.
+   */
+  private String preferencesNotificationCategoryKey(String identifier) {
+    return NOTIFICATION_CATEGORY_KEY_PREFIX + identifier;
```

Comment:
I think prefixing is not necessary. We use unique name for whole preferences, which is enough to avoid conflicts.

Reply:
Since we prefix the [notification requests](https://github.com/expo/expo/blob/master/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/service/SharedPreferencesNotificationsStore.java#L31), and use the same shared preferences file as `SharedPreferencesNotificationsStore`, I think we should keep the prefix? In case we add another store later, to avoid conflicts with future keys. 

If we do want to get rid of prefixing, maybe we should use a unique `SHARED_PREFERENCES_NAME`?

Reply:
Okay, you ar right. We might use another preference name, but keeping all preference-related info in one file makes sense.